packages/editor/src/components/document-outline/item.js

```diff
@@ -29,9 +29,9 @@ const TableOfContentsItem = ( {
 			}
 		) }
 	>
-		<button
+		<a
+			href={ target ? '#block-' + target : '#post-title-1' }
```

Comment:
The `-1` here on the post title is something I don't think we want to rely on. Since it's the incrementer from `withInstanceId`, there's the possibility that it could be changed to `-2` if the title was remounted for some reason (one I could imagine is the "Attempt Recovery" button of the catastrophic error handler).

Traditionally, this is where we'd employ either global state or another [strategy for communicating between any two components](https://www.javascriptstuff.com/component-communication/#the-8-strategies). It feels a bit heavy-handed in this instance.

There's a few other subtleties to this. The connection between title and a lack of a `target` value here is quite loose / implicit.

I'd be inclined to one of (in order of preference):

- Keep `onClick` for title focusing
- Make title entry in DocumentOutline unclickable
- Concede and assign a static `id` for title
- Create some arrangement of context or global app state for the outline to be aware of the correct instance of the title element

Reply:
> Concede and assign a static id for title

Can there ever be more than one title block? If there is only one title block, can we assign it the static id '#post-title'? Why is there an incrementer?

One other option: add a helper/selector that returns the correct/current title block id.

Reply:
>Can there ever be more than one title block?

The reason it's not a straight-cut answer is because:

- _Can_ there be? Yes, any React component can be used any number of times.
- _Will_ there be? No, not as implemented today.

Reply:
@aduth I addressed this in https://github.com/WordPress/gutenberg/pull/10815/commits/ef96b90139bb259b3b5d40be4d2e5be74badcdf1 by using the existing titleNode locator code, and passing the id to target.